### ISSUE TITLE
Has the development of the XUL add-on stopped in favor of the WebExtensions one?

### ISSUE THREAD

**Issue by jdunn0** [danny0838/firefox-scrapbook#282]
I'm curious if the two types of add-ons are being developed at the same time or if the WebExtension based one will be the only one with active development.
I ask because the XUL add-on is more flexible and has more features then the WebExtensions version and works just fairly well on Waterfox (multi-process needs to disabled), Pale Moon and Basilisk.
Personally, as Mozilla won't let you use have file system access (I read the Mozilla bug on allowing that where they pretty much said no) in a WebExtension, that would be the point where I just say forget it and use browsers made my people that actually seem to care about their developers and user.

**Comment by danny0838** [comment]
Yes, WebScrapBook is the only one in active development. This is because XUL/XPCOM is a very old technique using ECMAScript 3 spec causing the code extremely hard to be reworked to support increasing features in HTML5 or latest ECMAScript spec. Currently a web page captured using WebScrapBook is more faithful than legacy ScrapBook X, especially for web pages using modern techniques.

Although Firefox and Chromium does not provide direct javascript API for filesystem access, this can be workaround using a server script such as the current PyWebScrapBook, with whose help most features in legacy ScrapBook can be achieved (although the migration haven't completed yet and the performance is sub-optimal currently).

Market sharing is also an issue. WebScrapBook runs on both Firefox and Chromium based browsers, including mobile platforms, while browsers supporting legacy ScrapBook X is very restricted.
